src/main/java/entity/Item.java

```diff
@@ -0,0 +1,50 @@
+package entity;
+
+public abstract class Item {
+    private final String name; // is this final??? // the name of the entity.Item
+    private int num; // the number of items in entity.Item
+
+    /**
+     * Creates a new entity.Item object.
+     * This constructor takes in a single String as and a single int as arguments..
+     * @param name the name of the entity.Item.
+     * @param num the number of items in the entity.Item object.
+     */
+    public Item(String name, int num) {
+        this.name = name;
```

Comment:
here since you have a constructor taking the string name you can just write:
`this(name)`
which will call the constructor taking just a String, instead of rewriting the same code again